Sources/Deferred.swift

```diff
@@ -56,32 +62,27 @@ public struct Deferred<Value>: FutureType, PromiseType {
 
     // MARK: FutureType
 
-    private func upon(queue: dispatch_queue_t, options inOptions: dispatch_block_flags_t, body: Value -> Void) -> dispatch_block_t {
+    private func upon(executor: ExecutorType, options inOptions: dispatch_block_flags_t, body: Value -> Void) -> dispatch_block_t {
```

Comment:
Yeah, `inOptions` is definitely not in line. Will think about `body`'s external parameter.